server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -269,19 +271,32 @@ public Builder(
             DateFormatter defaultFormat = resolution == Resolution.MILLISECONDS
                 ? DEFAULT_DATE_TIME_FORMATTER
                 : DEFAULT_DATE_TIME_NANOS_FORMATTER;
-            this.format = Parameter.stringParam(
-                "format",
-                indexCreatedVersion.isLegacyIndexVersion(),
-                m -> toType(m).format,
-                defaultFormat.pattern()
-            );
+
+            if (indexCreatedVersion.major <= Version.CURRENT.previousMajor().major) {
```

Comment:
Should this version be a fixed constant? ie why not hardcoded to 8, or use one of the V_8_* constants (maybe 8.0.0?). Otherwise when the major version bumps this would start applying to 8.x.